fontbe/src/post.rs

```diff
@@ -44,13 +45,43 @@ impl Work<Context, AnyWorkId, Error> for PostWork {
             .global_metrics
             .get()
             .at(static_metadata.default_location());
-        let postscript_names = &static_metadata.postscript_names;
-        let glyph_order = context.ir.glyph_order.get();
-        let mut post = Post::new_v2(
-            glyph_order
-                .names()
-                .map(|g| postscript_names.get(g).unwrap_or(g).as_str()),
-        );
+        // Whether to rename glyphs for 'production' using the provided remame map
+        let do_production_names = context.flags.contains(Flags::PRODUCTION_NAMES);
+        let rename_map = &static_metadata.postscript_names;
+        let mut seen = HashMap::new();
+        let final_glyph_names: Vec<_> = context
+            .ir
+            .glyph_order
+            .get()
+            .names()
+            .map(|g| {
+                if !do_production_names {
+                    // pass through the original name
+                    return g.to_string();
```

Comment:
nit: we're now copying the names when the flag is off where previously we were able to use str and not copy? Maybe it would make sense to branch on the flag and call one of two functions to produce Post?

Reply:
yea I agree with something like this; we can just have a `make_prod_names` fn that we call if the flag is set. In general I'm also trying to have less stuff inside `exec`, because it makes it easier to write unit tests.